src/core/thread/mle_router.cpp

```diff
@@ -1042,26 +1046,23 @@ uint8_t MleRouter::LqiToCost(uint8_t aLqi)
 
 uint8_t MleRouter::GetLinkCost(uint8_t aRouterId)
 {
-    uint8_t rval;
+    uint8_t rval = kMaxRouteCost;
+    Router *router;
 
-    // kInvalidRouterId indicates non-existing next hop, hence return kMaxRouteCost for it.
-    assert(aRouterId <= kInvalidRouterId);
+    router = GetRouter(aRouterId);
 
-    VerifyOrExit(aRouterId != mRouterId &&
-                 IsRouterIdValid(aRouterId) &&
-                 mRouters[aRouterId].mState == Neighbor::kStateValid,
-                 rval = kMaxRouteCost);
+    // kInvalidRouterId indicates non-existing next hop, hence return kMaxRouteCost for it.
```

Comment:
Maybe change the `kInvalidRouterId indicates` comment to `NULL aRouterId indicated`.